packages/terra-navigation-prompt/docs/README.md

```diff
@@ -0,0 +1,138 @@
+
+The NavigationPrompt can be rendered with two props: a `description` string that gives context to the NavigationPrompt, and a `metaData` object that can be used to provide additional data that may be relevant to the NavigationPrompt.
+
+The NavigationPrompt can be rendered like any other component, but prop consistency should be maintained between renders when possible. Creating a new object for the `metaData` prop within a render function will cause the NavigationPrompt to deregister and reregister itself during every update:
+
+```jsx
+render() {
+  return (
+    <NavigationPrompt description="This prompt will register and notify every time the component renders" metaData={{ important: 'info' }} />
+  );
+}
+```
+
+To avoid this, the `metaData` value can be stored inside a component's state or an internal property and updated only when necessary:
+
+```jsx
+constructor() {
+  this.state = {
+    metaDataValue: {
+      important: 'info',
+    }
+  }
+}
+
+render() {
+  return (
+    <NavigationPrompt description="My props are consistent between renders, so this prompt will not re-register or notify" metaData={this.state.metaDataValue} />
+  );
+}
+```
+
+### `NavigationPromptCheckpoint`
+
+The NavigationPromptCheckpoint can be used to detect the presence of NavigationPrompts rendered within it. 
+
+#### `onPromptChange`
+
+The `onPromptChange` function prop is used to communicate NavigationPrompt registrations to the implementer of the NavigationPromptCheckpoint. The function is called with an array of objects containing each registered NavigationPrompt's `description` and `metaData` properties.
+
+> Note: The `onPromptChange` prop will be called after each NavigationPrompt registration and removal. It may be called more than once with the same array of NavigationPrompt data. Calls to `setState` within onPromptChange should be performed only when necessary to minimize component updates.
+
+#### `resolvePrompts`
+
+The `resolvePrompts` function be accessed from the ref to a NavigationPromptCheckpoint. Calling `resolvePrompts` results in a Promise being returned and a NotificationDialog being presented to the user with options to either confirm or cancel their action. If the user confirms the action, the dialog will close, and the returned Promise will be resolved. If the user cancels the action, the dialog will close, and the returned Promise will be rejected. If no NavigationPrompts are detected, no dialog is presented, and a resolved Promise is returned.
+
+`resolvePrompts` accepts a single Object argument that should contain the text strings used to render the NotificationDialog. If these strings need to present information based on the currently registered NavigationPrompts, the `onPromptChange` prop can be used to receive that data and build those strings as needed.
+
+The supported keys in the resolvePropts argument include:
+
+|Key|Type|Description|
+|---|---|---|
+|`title`|String|The title of the NotificationDialog|
```

Comment:
Do we provide default strings?

Reply:
There are no default strings. We could include them if we can land on a reasonable set of terms, but from what I can tell, the messaging will likely need to be specific to each use case.

Consumer's could wrap the NavigationPromptCheckpoint to provide their own default strings if that's desirable to them.